requests/models.py

```diff
@@ -399,7 +399,10 @@ def prepare_headers(self, headers):
         """Prepares the given HTTP headers."""
 
         if headers:
-            self.headers = CaseInsensitiveDict((to_native_string(name), value) for name, value in headers.items())
+            self.headers = CaseInsensitiveDict((
+                                to_native_string(name),
+                                convert_to_string(value)
+                            ) for name, value in headers.items())
```

Comment:
This is really weird at this point, but can we do this instead?

```python
self.headers = CaseInsensitiveDict(
    (to_native_string(name), convert_to_string(value))
    for name, value in headers.items()
)
```

This just reads a little bit better to me.